### ISSUE TITLE
isIPAddress has no unit tests

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2688]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2688|
|Type|defect|
|Created|2007-06-03 15:09:07Z|

<details><summary>Searchable metadata</summary>

```
trac-id__2688 2688
type__defect defect
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1180883347000000 1180883347000000
changetime__1181227972000000 1181227972000000
version__None None
owner__ 
cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun removed owner|
|-|-|

Done in isipaddress-tests-2688.  This is basically the branch from [#2654](https://github.com/twisted/twisted/issues/2654), but with the new behavior removed and only the improvements to existing functionality remaining.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve set owner to @exarkun|
|-|-|

Please add the test-case-name header to abstract.py, then merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

(In [20489]) Merge isipaddress-tests-2688

Author: exarkun
Reviewer: therve
Fixes [#2688](https://github.com/twisted/twisted/issues/2688)

Add unit tests for twisted.internet.abstract.isIPAddress and
improve exception handling in the implementation.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
